### PR TITLE
Feature best fit circle

### PR DESCRIPTION
**Pull request by CristianoPizzamiglio**
Add `best_fit `class method to the Circle class in \scikit-spatial\src\skspatial\objects\circle.py.

Add two tests for the method in \scikit-spatial\tests\unit\objects\test_circle.py. The tests are:
- `test_best_fit`
- `test_best_fit_failure`

Update api reference documentation (\scikit-spatial\docs\source\api_reference\skspatial.objects.Circle.rst).

### CODE REVIEW

File: docs/source/api_reference/skspatial.objects.Circle.rst [hunk 1]
```diff
@@ -13,4 +13,5 @@ Methods
    ~skspatial.objects.Circle.area
    ~skspatial.objects.Circle.circumference
    ~skspatial.objects.Circle.intersect_line
+   ~skspatial.objects.Circle.best_fit
```
Comment (ajhynes7) [inline docs/source/api_reference/skspatial.objects.Circle.rst:16]: Could we move this line up two so the lines are in alphabetical order?

**Review by ajhynes7 — CHANGES_REQUESTED** [review]
Looks good! You can also add your name to `CONTRIBUTORS.md` in this PR.

File: docs/source/api_reference/skspatial.objects.Circle.rst (hunk 1 above)
Comment (CristianoPizzamiglio) [reply docs/source/api_reference/skspatial.objects.Circle.rst]: For sure, done! Now the they are in alphabetical order.

I apologize, I can't find the file `CONTRIBUTORS.md`. Do I need to create one? Thanks!

Comment (ajhynes7) [reply docs/source/api_reference/skspatial.objects.Circle.rst]: My mistake, the file is `AUTHORS.rst`. You can add your name in the Contributors section.

Comment (CristianoPizzamiglio) [reply docs/source/api_reference/skspatial.objects.Circle.rst]: Done!

**Review by ajhynes7 — APPROVED** [review]

**Comment by ajhynes7** [comment]
Good job on this, thanks!

**Pull request merged**